### DISCUSSION TITLE
Best USB Zigbee Adapter?

### DISCUSSION THREAD

**Discussion by mircolino** [Koenkk/zigbee2mqtt#7079] [Devices]
I'm just getting started with zigbee2mqtt and looking to purchase the best USB zigbee adapter for my RPi4.
After some research I'm down to the following three choices:

- Electrolama zzh!
- Slaesh CC2652RB
- dresden elektronik ConBee II

What are you guys recommending?
Is the zzh! better than the Slaesh? I see it's using a slightly different TI chip.
I can get the ConBee II on Amazon tomorrow but how experimental is the support in zigbee2mqtt?
Other options I missed?

I currently have about 25 zigbee devices:

- Hue motion sensors
- SmartThings leak sensors
- SmartThings smart plugs
- Innr smart plugs
- Nyce contact sensors
- Yale lock

**Comment by Koenkk** [comment]
- ZZH and sleash are recommended (https://www.zigbee2mqtt.io/getting_started/what_do_i_need.html#supported-zigbee-adapter)
- Differences between them: see point 3 of https://www.zigbee2mqtt.io/information/supported_adapters.html#notes

**Reply by mircolino** [reply]
Thank you. As the sleash is not currently available, I just purchased a zzh on Tindie.

One quick question about LEDs. I read that the zzh has a general purpose LED which however is not functional when flashed with the coordinator firmware. The sleash has two LEDs (green and red from the pictures). Are those active when flashed with the coordinator firmware? And if so what kind of state do they provide?

**Reply by mircolino** [reply]
Also, since I'm in US, are the Tube's CC2652 based coordinators recommended as well? 
Given the shipping delays from EU, they would be much easier to get.

https://www.tubeszb.com/shop/coordinators/2

**Reply by Koenkk** [reply]
- For both the leds are not working, this might be solved in the next fw upgrade
- I don't know the Tube's adapter, so no clue.

**Reply by mircolino** [reply]
Thank you.

> I don't know the Tube's adapter, so no clue

The reason I asked is because on [his website](https://www.tubeszb.com/shop/coordinators/2) he claims they are supported by zigbee2mqtt

**Reply by jack9603301** [reply]
I think CC2530/2531 can be considered if you need cheaper equipment, in fact I choose CC2531.

**Reply by DubhAd** [reply]
> Thank you.
> 
> > I don't know the Tube's adapter, so no clue
> 
> The reason I asked is because on [his website](https://www.tubeszb.com/shop/coordinators/2) he claims they are supported by zigbee2mqtt

I know various people over on the HA, Zigbee2MQTT, and DigiBlur's Discords that use them with Z2M.

**Reply by WombatHollow** [reply]
Koenkk, your recommendation is now 4 years old and a lot has change.   Any update to recommended adaptors (particularly USB for use with Raspi3B+).  The ZBDongle-E I have does not work with latest Z2M on Bookworm, it freezes the RasPi Solid, unless I start Z2M, let it error then plug in the dongle!
Thanks

**Comment by mercenaruss** [comment]
Check out also [ZigStar](https://zig-star.com)

**Comment by knowic** [comment]
@Koenkk 
Hi, Koenk, 
Would you please help to add these adapters to the list of supported Zigbee adapter? 
[CC2652R Dongle](https://www.aliexpress.com/item/1005002823262979.html?spm=a2g0o.productlist.0.0.1a1640b82yeViq&algo_pvid=e01b1872-ca85-4814-971f-ce9b058855b8&algo_exp_id=e01b1872-ca85-4814-971f-ce9b058855b8-0&pdp_ext_f=%7B%22sku_id%22%3A%2212000022351543786%22%7D)
[CC2538+CC2592 Dongle](https://www.aliexpress.com/item/1005002809329614.html?spm=a2g0o.store_pc_allProduct.8148356.2.4d7f1012TTc3uX)
[CC2652P Dongle](https://www.aliexpress.com/item/1005003393047763.html?spm=a2g0o.productlist.0.0.5e6f107dvPTg3a&algo_pvid=c932a9d4-2522-4b73-a001-02609564422e&algo_exp_id=c932a9d4-2522-4b73-a001-02609564422e-1&pdp_ext_f=%7B%22sku_id%22%3A%2212000025577120008%22%7D)

Most of the dongle function is compatible with TI developments kit like LAUNCH-XL series with buttons/led/UART IOs,  more IOs is breaking out, cJTAG is availble for firmware burning without bootloader enable, and the RF  is tested and optimizated for best performance. The hardware is open-souce for users. 

 Here is some detailed information about it with CC2652R as example:
1. 2.4GHz multiprotocol RF development board(OpenThread,Zigbee,BLE,IEEE802.15.4,6LoWPAN and proprietary) based on TI CC2652R1FRGZR chip
2. Communicates with the host computer via the common CH340E USB-UART bridge
3. cJTAG debug header, flashed firmware or debug with JTAG
3. Self-programming via the TI serial bootloader with bootloader bit enabled
4. SMA antenna port for an external antenna with personal choice
5. General LEDs(Red--Power on indication  Green--DIO7 Blue--DIO6)
6. General Buttons(BTN1--DIO13<Bootloader-enable>  BTN2--DIO14 RST--nRESET)
7. IO  breakouts with 2.0mm pitch holes for connecting peripheral device
8. Low power LDOs
9. ESD protections with USB interface
10. easy to use with Koenkk's zigbee2mqtt
11. easy to use with TI's OpenThread border router
12. Confirmed working with ZHA,ZIGBEE2MQTT
13. Compatible with LAUCH-XL CC2652R1 from TI

thanks,

Jason

**Reply by Koenkk** [reply]
Please make a pr for https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/information/supported_adapters.md (click the pencil button)

**Reply by knowic** [reply]
Hi, Koenk,
I have create a pull request, but I haven't get to upload the picture ,  How to ? 

thanks again.
Jason

**Comment by alex-v-fraser** [comment]
@Koenkk  Hi! Yesterday I moved from Raspbee II to Sonoff Zigbee 3.0 USB Dongle Plus. I've got significant LQI decrease of all zigbee devices (first screenshot). For example: the "koridor presence sensor" (second screenshot) is in direct visibility zone at a distance of 30-40 cm from coordinator antenna. Transmit_power is changed to 19, but no result. Will you help me, please? How can I fix it? Use another stick?

![screen 1](https://user-images.githubusercontent.com/77446865/166237678-e5bfab0b-7190-4d0f-847d-2f29d7be5ff8.png)
![screen 2](https://user-images.githubusercontent.com/77446865/166237685-bb47dd60-2676-47c7-b4b1-dc382a57fb94.png)
